src/main/java/com/squareup/javapoet/TypeSpec.java

```diff
@@ -262,7 +274,13 @@ void emit(CodeWriter codeWriter, String enumName, Set<Modifier> implicitModifier
         Collections.<Modifier>emptySet(),
         Collections.<Modifier>emptySet(),
         Collections.<Modifier>emptySet(),
-        Collections.singleton(Modifier.STATIC));
+        Collections.singleton(Modifier.STATIC)),
+
+    ANNOTATION(
```

Comment:
I think it's a bug in our code that `Kind` is a private interface, since we expose it in the public `Kind` field. Do you mind making `Kind` public, and then we don't need `isAnnotationDeclaration()` ? The caller can just check the kind!